### ISSUE TITLE
Relax error fatalness on too much recursion or maximum call stack

### ISSUE THREAD

**Issue by leopsidom** [pyodide/pyodide#1605]
Picking up the failed example posted [here](https://github.com/pyodide/pyodide/issues/821#issuecomment-831456556):

```
import jedi; import numpy; jedi.Interpreter("numpy.in1d([1,2,3],[2]).", [globals()]).complete()
```

Currently this will give `too much recursion` error on firefox and `maximum call stack` error on chrome and throw a fatal error from pyodide.

> Pyodide has suffered a fatal error. Please report this to the Pyodide maintainers.

And leave pyodide session in an unusable state. Is it possible to only throw an exception in such cases but still keep pyodide session alive ? I think this would be pretty useful in a console environment where we don't want one failure to kill the entire session.

**Comment by hoodmane** [comment]
If you were running in a native Python, this would be a segfault. After this error, Pyodide will display unstable behavior. I think it's better to lock people out than to keep going but sometimes do weird stuff.

Normally when there is an exception, Python would handle it, unwind the stack, reset state. When one of these low level JS/Wasm errors occurs, it interrupts Python main loop in the middle of whatever is happening, potentially leaving the Python VM in an invalid state. This means that the Python VM is unrecoverable. Like a segfault.

**Comment by leopsidom** [comment]
Okay got it, that makes sense. Wonder though, if we can give a fatal error on the potentially unstable behavior the session can have but still let user decide if they want to start a new session or keep using it. Or have some options to configure the behavior through pyodide. I think people can have more tolerance when they are using a console.

**Comment by hoodmane** [comment]
> I think people can have more tolerance when they are using a console.

I agree, I can see benefit to allowing an override: "Okay, keep going but if weird stuff starts happening, we warned you!"

**Comment by hoodmane** [comment]
Closing as resolved due to improved recursion limit since v0.19. Please open a new issue if you still think there is a problem.
